File: vama-frontend/src/types/category.ts
export interface Category {
  id: string
  name: string
  slug: string
  parentId: string | null
  description?: string
  productCount: number
  isActive: boolean
  createdAt: string
  updatedAt: string
}

export type CreateCategoryDto = Pick<Category, 'name' | 'parentId' | 'description' | 'isActive'>
export type UpdateCategoryDto = Partial<CreateCategoryDto>

File: vama-backend/src/main/java/com/vama/catalog/domain/Category.java
package com.vama.catalog.domain;

import com.vama.common.model.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "categories")
public class Category extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String slug;

    private UUID parentId;

    private String description;

    @Column(nullable = false)
    private boolean isActive;
}
